includes/api/class-wc-admin-rest-reports-performance-indicators-controller.php

```diff
@@ -0,0 +1,268 @@
+<?php
+/**
+ * REST API Performance indicators controller
+ *
+ * Handles requests to the /reports/store-performance endpoint.
+ *
+ * @package WooCommerce Admin/API
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST API Reports Performance indicators controller class.
+ *
+ * @package WooCommerce/API
+ * @extends WC_REST_Reports_Controller
+ */
+class WC_Admin_REST_Reports_Performance_Indicators_Controller extends WC_REST_Reports_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v3';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'reports/performance-indicators';
+
+	/**
+	 * Maps query arguments from the REST request.
+	 *
+	 * @param array $request Request array.
+	 * @return array
+	 */
+	protected function prepare_reports_query( $request ) {
+		$args           = array();
+		$args['before'] = $request['before'];
+		$args['after']  = $request['after'];
+		$args['stats']  = $request['stats'];
+		return $args;
+	}
+
+	/**
+	 * Get all allowed stats that can be returned from this endpoint.
+	 *
+	 * @return array
+	 */
+	public function get_allowed_stats() {
+		global $wp_rest_server;
+
+		$request       = new WP_REST_Request( 'GET', '/wc/v3/reports' );
+		$response      = rest_do_request( $request );
+		$endpoints     = $response->get_data();
+		$allowed_stats = array();
+		if ( 200 !== $response->get_status() ) {
+			return new WP_Error( 'woocommerce_reports_performance_indicators_result_failed', __( 'Sorry, fetching performance indicators failed.', 'wc-admin' ) );
+		}
+
+		foreach ( $endpoints as $endpoint ) {
+			if ( '/stats' === substr( $endpoint['slug'], -6 ) ) {
+				$request  = new WP_REST_Request( 'OPTIONS', '/wc/v3/reports/' . $endpoint['slug'] );
+				$response = rest_do_request( $request );
+				$data     = $response->get_data();
+				$prefix   = substr( $endpoint['slug'], 0, -6 );
+
+				if ( empty( $data['schema']['properties']['totals']['properties'] ) ) {
+					continue;
+				}
+
+				foreach ( $data['schema']['properties']['totals']['properties'] as $property_key => $schema_info ) {
+					$allowed_stats[] = $prefix . '/' . $property_key;
+				}
+			}
+		}
+		return $allowed_stats;
```

Comment:
Should this have a filter on it to allow for extensions to register allowed_stats from other non `/wc/v3/reports` endpoints - or possibly to exclude stats from some users?